### PR TITLE
test: add tests for associated type `Record`

### PR DESCRIPTION
**Pull request by jaschdoc**
Supersedes https://github.com/mlutze/flix/pull/54

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@herluf-ba These tests reveal some issues in the new Weeder :)

**Comment by herluf-ba** [comment]
I made a pr that should fix the issue now 👍 I want to make you aware that the test is still failing due to `ConstraintResolver`. That's intended?

**Comment by magnus-madsen** [comment]
> I made a pr that should fix the issue now 👍 I want to make you aware that the test is still failing due to `ConstraintResolver`. That's intended?

Well, the tests are to find bugs in the ConstraintResolver, so I guess the answer is both yes and no :)

**Comment by jaschdoc** [comment]
Still relevant: https://github.com/mlutze/flix/pull/54#issuecomment-2020524706

**Comment by mlutze** [comment]
!!!rerun

File: main/test/flix/Test.Dec.AssocType.Record.flix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+    instance R[Int32] {
+        type K = { x = Int64, x = Int32 }
+        pub def f(x: Int32): { x = Int64, x = Int32 } = { x = Int32.toInt64(x), x = x }
+    }
+
+    instance R[Vector[a]] with R[a] {
+        type K = { x = Int32, y = Vector[R.K[a]] }
+        pub def f(x: Vector[a]): { x = Int32, y = Vector[a] } = { x = Vector.length(x), y = x }
```
Comment (mlutze) [inline main/test/flix/Test.Dec.AssocType.Record.flix:19]: This the return type is wrong. It should be the same as K. i.e., `{ x = Int32, y = Vector[R.K[a]] }`

**Comment by mlutze** [comment]
Is this still broken?

**Comment by jaschdoc** [comment]
All good now. The error from the previous pr throws a type error instead of crashing :)

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**